tests/unit/igraph_compare_communities.c

```diff
@@ -0,0 +1,102 @@
+/*
+   IGraph library.
+   Copyright (C) 2021  The igraph development team <igraph@igraph.org>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include <igraph.h>
+#include "test_utilities.inc"
+
+char *names[] = {
+    [IGRAPH_COMMCMP_VI] = "VI",
```

Comment:
Hmmm, this is nice but I wonder whether this construct is understood by most C compilers. Is this a standard thing? @szhorvat ?

Reply:
Designated initializers are in C99, not in earlier standards. I hope I remember correctly we use C99.

Figuring out if MSVC actually supports it is hard, but I found a blog post comment which references another comment that doesn't exist anymore which seems to indicate that VS 2013 supports it.

https://devblogs.microsoft.com/cppblog/c11-and-c17-standard-support-arriving-in-msvc/#comment-1230

I used it to not depend on the enum order.

Reply:
I tried it here with VS2015, seems to be good: https://godbolt.org/z/z3qWa1

Reply:
Note that it's only available in C mode, not in C++ mode, but that's fine.

We should keep supporting VS2015, i.e. version 19.0 (not 19.x where x > 0), for the time being. This seems to be fine with 19.0